mlflow/langchain/_langchain_autolog.py

```diff
@@ -257,13 +257,30 @@ def _inject_callbacks(original_callbacks, new_callbacks):
     """
     from langchain_core.callbacks.base import BaseCallbackManager
 
+    def _is_new_callback_already_in_original_callbacks(new_callback, original_callbacks):
+        return any(
+            isinstance(new_callback, type(original_callback))
+            for original_callback in original_callbacks
+        )
+
     if isinstance(original_callbacks, BaseCallbackManager):
         for callback in new_callbacks:
-            original_callbacks.add_handler(callback)
+            if not _is_new_callback_already_in_original_callbacks(
+                callback, original_callbacks.handlers
+            ):
+                original_callbacks.add_handler(callback)
```

Comment:
We shouldn't be modifying `original_callbacks` either. Filed a follow-up ticket to fix that.